client/components/has-site-purchases-switch/test/index.jsx

```diff
@@ -0,0 +1,82 @@
+/**
+ * @jest-environment jsdom
+ */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import '@testing-library/jest-dom/extend-expect';
+
+/**
+ * Internal dependencides
+ */
+import { render as rtlRender, screen } from 'config/testing-library';
+import HasSitePurchasesSwitch from '../index';
+import { reducer as purchases } from 'calypso/state/purchases/reducer';
+
+jest.mock( 'calypso/components/data/query-site-purchases', () => () => <p>Query</p> );
```

Comment:
It was easier here to mock the query component rather than mocking the endpoint.